java/src/test/java/org/apache/arrow/gandiva/evaluator/NativeEvaluatorTest.java

```diff
@@ -205,13 +152,15 @@ public void testEvaluate() throws GandivaException, Exception {
     for (int i = 8; i < 16; i++) {
       assertTrue(intVector.isNull(i));
     }
+
+    // free buffers
+    releaseArrowBufs(batch.getBuffers());
```

Comment:
- looks like these 3 lines are needed for every test. any chance we can do it automatically (AutoCloseable) ?
- how did you figure out the leak ? Does allocator.close() print a warning/error regarding the leak ? If yes, can you please add that too at the end of each test ?